components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/model/CarbonOAuthAuthzRequest.java

```diff
@@ -80,4 +85,25 @@ protected OAuthValidator<HttpServletRequest> initValidator() throws OAuthProblem
 
         return OAuthUtils.instantiateClass(clazz);
     }
+
+    @Override
+    public String getState() {
```

Comment:
Move the logic to error handling place. Use nimbus library to decode req obj

Reply:
fixed with ef0e046274f92b950041433c237956dc64e81944